File: app/src/main/java/com/lucciola/haskellanywhereforandroid/controller/SendButtonController.java
package com.lucciola.haskellanywhereforandroid.controller;

import com.lucciola.haskellanywhereforandroid.View;
import com.lucciola.haskellanywhereforandroid.ProgramManager;
import com.lucciola.haskellanywhereforandroid.uiaction.UIAction;
import com.lucciola.haskellanywhereforandroid.uiaction.UIActionFactory;

/**
 * Created by hiden on 2017/06/10.
 */

public class SendButtonController extends Controller {

    public SendButtonController(View arg0, ProgramManager arg2) {
        super(arg0, arg2);
    }

    @Override
    public void onClick(android.view.View v) {
        String input = this.view.getInputText();
        this.view.clearInputText();
        UIAction uiAction = UIActionFactory.createUIAction(manager.input(input), view);
        uiAction.execute();
    }

}

File: app/src/main/java/com/lucciola/haskellanywhereforandroidkt/controller/SendButtonController.kt
package com.lucciola.haskellanywhereforandroidkt.controller

import com.lucciola.haskellanywhereforandroidkt.View
import com.lucciola.haskellanywhereforandroidkt.ProgramManager
import com.lucciola.haskellanywhereforandroidkt.uiaction.UIAction
import com.lucciola.haskellanywhereforandroidkt.uiaction.UIActionFactory

/**
 * Created by hiden on 2017/06/10.
 */

class SendButtonController(arg0: View, arg2: ProgramManager) : Controller(arg0, arg2) {

    override fun onClick(v: android.view.View) {
        val input = this.view.inputText
        this.view.clearInputText()
        val uiAction = UIActionFactory.createUIAction(manager.input(input), view)
        uiAction!!.execute()
    }

}
